src/price_data.py

```diff
@@ -563,3 +568,45 @@ def get_cost(
         if isinstance(tr, transaction.SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+
+    def check_database(
+        self,
+    ):
+        for db_path in Path(config.DATA_PATH).glob("*.db"):
+            if db_path.is_file():
+                platform = db_path.stem
+
+                try:
+                    get_price = getattr(self, f"_get_price_{platform}")
+                except AttributeError:
+                    if platform == "coinbase":
+                        get_price = self._get_price_coinbase_pro
+                    else:
+                        raise NotImplementedError(
+                            "Unable to read data from %s", platform
+                        )
+
+                with sqlite3.connect(db_path) as conn:
+                    query = "SELECT name FROM sqlite_master where type='table'"
+                    cur = conn.execute(query)
+
+                    for table in cur.fetchall():
```

Comment:
`table` is a confusing variable name. Perhaps call it `result` or something like that instead.

Or use a generator expression `tablenames = (r[0] for r in cur.fetchall()); for tablename in tablenames ...`